src/key/helper.js

```diff
@@ -38,17 +38,29 @@ export async function generateSecretKey(options) {
  */
 export async function getLatestValidSignature(signatures, primaryKey, signatureType, dataToVerify, date = new Date()) {
   let signature;
+  let exception;
   for (let i = signatures.length - 1; i >= 0; i--) {
-    if (
-      (!signature || signatures[i].created >= signature.created) &&
-      // check binding signature is not expired (ie, check for V4 expiration time)
-      !signatures[i].isExpired(date) &&
-      // check binding signature is verified
-      (signatures[i].verified || await signatures[i].verify(primaryKey, signatureType, dataToVerify))
-    ) {
-      signature = signatures[i];
+    try {
+      if (
+        (!signature || signatures[i].created >= signature.created) &&
+        // check binding signature is not expired (ie, check for V4 expiration time)
+        !signatures[i].isExpired(date) &&
+        // check binding signature is verified
+        (signatures[i].verified || await signatures[i].verify(primaryKey, signatureType, dataToVerify))
+      ) {
+        signature = signatures[i];
+      }
+    } catch (e) {
+      exception = e;
```

Comment:
If there is array of signatures, than there could be an array of exceptions?

Reply:
Theoretically yes, but there's a lot of other places in the library where we only give one error even if multiple can occur, e.g. https://github.com/openpgpjs/openpgpjs/blob/9bdeaa927ace1ef32ce025140651966e2e7751eb/src/key/key.js#L378-L389
I think in practice it's uncommon enough that it's not worth the effort to concatenate all the error messages or something. If someone wants more info they can debug the key manually.